clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -1073,6 +1074,8 @@ void Clang::AddPreprocessingOptions(Compilation &C, const JobAction &JA,
     getToolChain().AddCudaIncludeArgs(Args, CmdArgs);
   if (JA.isOffloading(Action::OFK_HIP))
     getToolChain().AddHIPIncludeArgs(Args, CmdArgs);
+  if (JA.isOffloading(Action::OFK_SYCL))
+    toolchains::SYCLToolChain::AddSYCLIncludeArgs(D, Args, CmdArgs);
```

Comment:
Why can we not do a 'getToolChain() for SYCL? Thanks

Reply:
We can probably match up here, the use of getToolChain() is dependent on the given toolchain so it should be fine as we are in the SYCLToolChain here.

Reply:
I have made some significant updates here to sync things up a bit better to allow for `getToolChain()` usage.